src/main/python/org/broadinstitute/hellbender/gcnvkernel/inference/deterministic_annealing.py

```diff
@@ -44,12 +50,70 @@ def __init__(self,
                  temperature=None):
 
         assert temperature is not None, "Temperature (a scalar theano shared tensor) is not provided"
+        approx = DeterministicMeanField(local_rv=local_rv,
+                                        model=model,
+                                        cost_part_grad_scale=cost_part_grad_scale,
+                                        scale_cost_to_minibatch=scale_cost_to_minibatch,
+                                        random_seed=random_seed,
+                                        start=start)
         super().__init__(
-            KLThermal, MeanField, None,
-            local_rv=local_rv,
-            model=model,
-            cost_part_grad_scale=cost_part_grad_scale,
-            scale_cost_to_minibatch=scale_cost_to_minibatch,
-            random_seed=random_seed,
-            start=start,
-            op_kwargs={'temperature': temperature})
+            KLThermal, approx, None, temperature=temperature)
+
+class DeterministicApproximation(Approximation):
```

Comment:
Some comments here about the determinism issues are definitely merited---summarize the problem in the original code and explain the fix here. Maybe also link the Slack thread---perhaps just on Github, rather than in code comments?

Reply:
https://broadinstitute.slack.com/archives/C1HH1V5EC/p1663790673088429